docs/src/services/loki.md

```diff
@@ -0,0 +1,113 @@
+# Loki
+
+Loki is our central log server that aggregates all logs we want to
+have in one place.  Note that this is not necessarily all logs, as
+Loki is queryable by anyone on the internet, and we don't want to push
+all syslog data to Loki.
+
+Loki is from Grafana, and there are two main ways to interact with it,
+both of which use LogQL.  If you are familiar with Prometheus LogQL
+should be fairly straightforward as it follows the same general design
+as PromQL.
+
+## PromQL Cheat Sheet
+
+PromQL expressions are always started by a stream selector in curly
+braces.  The stream selector must specify at least one stream by the
+labels that are available.  By default the following labels are
+available for all tasks:
+
+  - `nomad_job`: The top level ID of the job that is running.
+  - `filename`: The filename that the log was read from.
+
+Here's an example LogQL query that gives you the contents of the
+buildsync-musl log:
+
+```
+{nomad_job="buildsync-musl"}
+```
+
+This will pull all streams that are labelled with buildsync-musl as
+the Nomad job.  If you wanted only the stderr output, you could ammend
+the stream selector as follows:
+
+```
+{nomad_job="buildsync-musl", filename="/alloc/logs/rsync.stderr.0"}
+```
+
+Sometimes you might want to use regex to match multiple labels at once
+in the stream, such as if you want to get many logs from a logrotated
+set of files.  By default Nomad rotates its logs every 10MB.
+
+```
+{nomad_job=~"buildsync-(musl|aarch64)", filename=~"/alloc/logs/rsync.std(err|out).*"}
+```
+
+As shown above its possible to use regular expressions to match
+multiple log streams at once.  The operators for matching are as
```

Comment:
The above example demonstrates the use of a regular expression to also match multiple log streams simultaneously.